### DISCUSSION TITLE
How do you report an error in the page?

### DISCUSSION THREAD

**Discussion by steveculshaw** [PDConSec/vsc-print#307] [Q&A]
How do you report an error in the page?

The link to the manual is broken  :-(

**Comment by PeterWone** [comment]
Thanks for letting me know. You're actually the second person to report this in 24 hours which is quite gratifying because it means people actually do read the manual. 

Our release pipeline was set up when Node 14 was new and it worked so well that I had years to forget how it works. Then GitHub retired Node14 and things have been broken ever since. I thought I had it sorted out but my merge into the master branch yesterday changed the manual URL but the release failed. I've put a copy of the manual on the original path as a temporary fix.

**Reply by steveculshaw** [reply]
Many thanks for a great extension

**Reply by PeterWone** [reply]
Steve, 

If you have the time and inclination, I'd like to know how you use the tool. People don't answer "please fill in this survey" because, well, why _should_ you do unpaid corporate product research?

I built this because I needed it, and then I shared it and it grew.

It's actually quite elaborate these days - it's even possible for another extension to contribute a document renderer for a new format, and so far this supports source code, plaintext and rendered Markdown.  

Bizarrely some schools seem to want assignment source code as PDFs, so the support for printing a whole folder tree of source as a single document was a smash hit.

Anyhow... what is Print to you?

**Reply by steveculshaw** [reply]
Peter,
I'm a simple user ... I'm just typing txt files for keeping notes
That said, reading the manual, the Markdown functionality looks interesting

**Reply by PeterWone** [reply]
Rendered Markdown is becoming difficult due to the attitude of the VS Code team that the internal rendering pipeline is not intended for third-party use so there's nothing wrong with unannounced breaking changes. I used the internal rendering pipeline in order to pick up any Markdown extensions installed by the user, but it's at the point where this isn't worth the problems. I have a plan for a complete replacement.

While there are quite a few ducks I must line up first, you might be interested in one of my related projects. In a nutshell I want to treat the creation of a large structured document like a textbook the same way I would the source code of an application: multiple files in a folder tree with config files and a project file tying it all together. Compiling this project generates the index and ToC and renders the whole thing as a single HTML file with supporting resources.

Treating it as source code makes it amenable to version control with multiple concurrent contributors.

**Reply by steveculshaw** [reply]
Wow ... that sounds really neat
Documentation source control is always a pain
